tests/flux-v2-image-automation.sh

```diff
@@ -56,7 +62,54 @@ for FILE_LOCATION in $(echo ${FILE_LOCATIONS}); do
                 echo "Non whitelisted pattern found in ImagePolicy: $IMAGE_POLICY it should be ^prod-[a-f0-9]+-(?P<ts>[0-9]+)" && exit 1
             fi
         done
+    done
 
+    ##############################################################################################################
+    # Print success if ALL image policies are clean
+    ##############################################################################################################
+    printf "\n\n########## Image Policy documents checked and passing ########## \n\n"
+
+    ##############################################################################################################
+    # This section compiles a list of files that contains `image:` and stores them in an array HELMRELEASES
+    # Only files that follow these rules will be added to the array:
+    # - Contains the `image:` string
+    # - Is HelmRelease type document
+    ##############################################################################################################
+
+    HELMRELEASES=()
+    for FILE in $(grep -lr "image:" $FILE_LOCATION | grep -Ev "$EXCLUSIONS" ); do
+        while read -r doc; do
+            if [ "$doc" == "HelmRelease" ]; then
+                IFS=$'\n'
+                HELMRELEASES+=("$FILE")
+            fi
+        done < <(yq eval '.kind' $FILE)
     done
 
+    OUTPUTFILE="images.yaml"
+
+    for dir in $DIRECTORIES; do
+        exclude=false
+        for EXCLUDED_PATH in "${HELMRELEASES[@]}"; do
+        if [[ -f "$dir/$(basename $EXCLUDED_PATH)" ]]; then
+                exclude=true
+                break
+            fi
+        done
+    if ! $exclude; then
+        for RELEASE in "${HELMRELEASES[@]}"; do
+            nodejs_image=$(yq eval 'select(.spec.values.image) or (.spec.values.*.image) != null' $RELEASE)
```

Comment:
nodejs_image is no longer valid, should just be image/image_tag etc as its searching for all options now.